op-service/eth/transactions_test.go

```diff
@@ -30,111 +30,105 @@ func (m *MockL1Client) HeaderByNumber(ctx context.Context, number *big.Int) (*ty
+			name:             "nonceDiff_midRange",
+			currentBlock:     500,
+			blockConfirms:    5,
+			expectedBlockNum: 497,
+			expectedFound:    true,
+			nonces: map[uint64]uint64{
+				495: 5,
+				496: 5,
+				497: 6,
+				498: 6,
+				499: 6,
+				500: 6,
+			},
 		},
 		{
-			// Blocks       495 496 497 498 499 500
-			// Nonce          6   6   6   6   6   6
-			// call NonceAt   x   -   -   -   -   x
-			name:               "NonceUnchanged",
-			currentBlock:       500,
-			blockConfirms:      5,
-			previousNonceBlock: 400,
-			expectedBlockNum:   495,
-			expectedFound:      false,
+			name:             "nonceDiff_upperBound",
+			currentBlock:     500,
+			blockConfirms:    5,
+			expectedBlockNum: 500,
+			expectedFound:    true,
+			nonces: map[uint64]uint64{
+				495: 5,
+				496: 5,
+				497: 5,
+				498: 5,
+				499: 5,
+				500: 6,
+			},
+		},
+		{
+			name:             "nonce_unchanged",
+			currentBlock:     500,
+			blockConfirms:    5,
+			expectedBlockNum: 495,
+			expectedFound:    false,
+			nonces: map[uint64]uint64{
+				495: 6,
+				496: 6,
+				497: 6,
+				498: 6,
+				499: 6,
+				500: 6,
+			},
+		},
+		{
+			name:             "reorg",
+			currentBlock:     500,
+			blockConfirms:    5,
+			expectedBlockNum: 495,
+			expectedFound:    false,
+			nonces: map[uint64]uint64{
+				495: 5,
+				496: 7,
+				497: 6,
+				498: 6,
+				499: 6,
+				500: 6,
+			},
 		},
 	}
 
 	for _, tt := range tests {
-
 		t.Run(tt.name, func(t *testing.T) {
 			l1Client := new(MockL1Client)
 			ctx := context.Background()
 
-			currentNonce := uint64(6)
-			previousNonce := uint64(5)
-
+			// Setup mock responses
 			l1Client.On("HeaderByNumber", ctx, (*big.Int)(nil)).Return(&types.Header{Number: big.NewInt(int64(tt.currentBlock))}, nil)
-
-			// Setup mock calls for NonceAt, depending on how many times its expected to be called
-			if tt.previousNonceBlock < tt.currentBlock-tt.blockConfirms {
-				l1Client.On("NonceAt", ctx, common.Address{}, big.NewInt(int64(tt.currentBlock))).Return(currentNonce, nil)
-				l1Client.On("NonceAt", ctx, common.Address{}, big.NewInt(int64(tt.currentBlock-tt.blockConfirms))).Return(currentNonce, nil)
-			} else {
-				for block := tt.currentBlock; block >= (tt.currentBlock - tt.blockConfirms); block-- {
-					blockBig := big.NewInt(int64(block))
-					if block > (tt.currentBlock-tt.blockConfirms) && block < tt.previousNonceBlock {
-						t.Log("skipped block: ", block)
-						continue
-					} else if block <= tt.previousNonceBlock {
-						t.Log("previousNonce set at block: ", block)
-						l1Client.On("NonceAt", ctx, common.Address{}, blockBig).Return(previousNonce, nil)
-					} else {
-						t.Log("currentNonce set at block: ", block)
-						l1Client.On("NonceAt", ctx, common.Address{}, blockBig).Return(currentNonce, nil)
-					}
-				}
+			for blockNum, nonce := range tt.nonces {
+				l1Client.On("NonceAt", ctx, common.Address{}, new(big.Int).SetUint64(blockNum)).Return(nonce, nil)
```

Comment:
if you make the map key an `int64` you could use this slightly shorter form
```suggestion
				l1Client.On("NonceAt", ctx, common.Address{}, big.NewInt(blockNum).Return(nonce, nil)
```